static/js/publisher/release/components/defaultTrackModifier.js

```diff
@@ -0,0 +1,220 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
+
+import { getTracks, getTrackRevisions } from "../selectors";
+import { openModal, closeModal } from "../actions/modal";
+import { showNotification, hideNotification } from "../actions/notification";
+import { setDefaultTrack } from "../actions/defaultTrack";
+
+class DefaultTrackModifier extends Component {
+  constructor(props) {
+    super(props);
+
+    this.setDefaultTrackHandler = this.setDefaultTrackHandler.bind(this);
+    this.handleCloseModal = this.handleCloseModal.bind(this);
+    this.confirmDefaultTrackHandler = this.confirmDefaultTrackHandler.bind(
+      this
+    );
+    this.clearDefaultTrackHandler = this.clearDefaultTrackHandler.bind(this);
+    this.confirmClearDefaultTrackHandler = this.confirmClearDefaultTrackHandler.bind(
+      this
+    );
+  }
+
+  componentDidUpdate(prevProps) {
+    const {
+      defaultTrack,
+      currentTrack,
+      closeModal,
+      showNotification,
+      snapName
+    } = this.props;
+
+    if (defaultTrack.track !== prevProps.defaultTrack.track) {
+      closeModal();
+
+      if (defaultTrack.track === null) {
+        showNotification({
+          status: "success",
+          appearance: "positive",
+          content: `The default track for ${snapName} has been removed. All new installations without a specified track (e.g. \`sudo snap install ${snapName}\`) will receive updates from latest track.`,
+          canDismiss: true
+        });
+      } else {
+        showNotification({
+          status: "success",
+          appearance: "positive",
+          content: `The default track for ${snapName} has been set to ${currentTrack}. All new installations without a specified track (e.g. \`sudo snap install ${snapName}\`) will receive updates from the newly defined default track ${currentTrack}.`,
+          canDismiss: true
+        });
+      }
+    }
+  }
+
+  handleCloseModal() {
+    const { closeModal } = this.props;
+
+    closeModal();
+  }
+
+  confirmDefaultTrackHandler() {
+    const { currentTrack, setDefaultTrack, csrfToken, snapName } = this.props;
+
+    setDefaultTrack(snapName, csrfToken, currentTrack);
```

Comment:
It seems that all the params for `setDefaultTrack` are already in the store, so the action creator function could access them via `getState`. No need to pass them through props and to action.

So I guess we could have `setDefaultTrack()` - that sets current track as default and `resetDefaultTrack` - that sets it to null.